### PR TITLE
fix(ai-sidecar): Remove session/close capability advertisement

### LINKED ISSUE #8738: Gemini sidecar advertises session/close but its router rejects the call
## Symptom

Every chat request leaves a traceback in the Gemini sidecar log:

```
[sidecar] UserWarning: The method session/close is part of the unstable protocol, please enable `use_unstable_protocol` flag to use it.
[sidecar]   result = await self._handler(method, message.get("params"), False)
[sidecar] ERROR root: Background task failed
[sidecar] Traceback (most recent call last):
[sidecar]   File ".venv/.../acp/task/supervisor.py", line 51, in _on_done
[sidecar]     task.result()
[sidecar]   File ".venv/.../acp/task/dispatcher.py", line 81, in runner
[sidecar]     result = await self._request_runner(message)
[sidecar]   File ".venv/.../acp/connection.py", line 216, in _run_request
[sidecar]     result = await self._handler(method, message.get("params"), False)
[sidecar]   File ".venv/.../acp/router.py", line 184, in __call__
[sidecar]     return await route.handle(params)
[sidecar]   File ".venv/.../acp/router.py", line 72, in handle
[sidecar]     raise RequestError.method_not_found(self.method)
[sidecar] acp.exceptions.RequestError: Method not found
```

## Root cause

Two-layer mismatch between the sidecar and the upstream `agent-client-protocol` Python SDK:

1. **`scripts/ai-sidecar/gemini/sidecar.py:118-120`** — `JaegerSidecarAgent.initialize` advertises `SessionCapabilities(close=SessionCloseCapabilities())`. Per `cmd/jaeger/.../jaegerai/handler_test.go:655`, the Go gateway sends `session/close` **only when** the agent advertises this capability (`TestChatHandlerSessionCloseFiresWhenAgentAdvertisesCapability` covers it).
2. The ACP Python SDK has since moved `session/close` behind `use_unstable_protocol=True`. The sidecar doesn't set the flag (`run_agent` defaults `use_unstable_protocol=False`), so its router doesn't bind `JaegerSidecarAgent.close_session`. Every gateway-issued `session/close` falls through to the router's default and gets rejected with `Method not found`.

**Net effect:** the `close_session` method at `sidecar.py:157` is dead code. It pops `_contextual_tools` for the session id — but the router never reaches it, so the pop never runs via this path.

Note that **`prompt`'s `finally` block at `sidecar.py:432-440` already does the same pop**, so the cleanup itself isn't missed in the common case. `close_session` was just the backup path for "client disconnects after `new_session` but before `session/prompt`" — currently broken regardless.

### Option A: Drop the capability advertisement + the close_session method (recommended)

Smallest change. The sidecar stops claiming a feature it can't deliver. The Go gateway sees no `SessionCloseCapabilities` and skips sending `session/close` — `TestChatHandlerSessionCloseFiresWhenAgentAdvertisesCapability` already verifies this gating, so the existing Go test still covers the right behavior.

What we lose: the backup cleanup for the "new_session without prompt" edge case. Operator impact is bounded (one stale contextual-tools entry per orphan session until process restart; the entry is a small dict).

```python
# sidecar.py:initialize
return InitializeResponse(
    protocol_version=PROTOCOL_VERSION,
    agent_capabilities=AgentCapabilities(),  # drop SessionCapabilities
    agent_info=Implementation(...),
)
```

…and delete `JaegerSidecarAgent.close_session` (and the `CloseSessionResponse`/`SessionCapabilities`/`SessionCloseCapabilities` imports).

### Option B: Enable `use_unstable_protocol=True`

Re-routes `session/close` to our existing `close_session` handler. The advertisement + method continue to work as designed.

```python
# In _run_agent_with_cleanup (or wherever AgentSideConnection is constructed):
conn = AgentSideConnection(agent, agent_writer, agent_reader,
                           listening=False,
                           use_unstable_protocol=True)
```

What we accept: opting into the *entire* unstable surface of the SDK, not just `session/close`. Future SDK releases may land breaking changes behind this flag with no warning to us.

## Recommendation

Option A. The backup cleanup it removes is rarely exercised; we already get the common-case cleanup from `prompt`'s `finally`. Avoids tying the sidecar to an unstable-protocol opt-in.

## Acceptance criteria

- A chat request that completes normally produces zero "Background task failed" / "Method not found" tracebacks in the sidecar log.
- Existing `TestChatHandlerSessionCloseFiresWhenAgentAdvertisesCapability` continues to pass (it's about gateway behavior given an agent that does advertise; the Gemini sidecar simply won't be such an agent any more under Option A).
- `test_complete_acp_workflow_with_fake_agent` continues to pass (the test's `FakeAgent` already constructs `AgentCapabilities()` without sessions caps, so it's unaffected).

### PR DESCRIPTION
**Pull request by qsxDree**
## Which problem is this PR solving?
- Fixes #8738

## Description of the changes
- Remove `SessionCapabilities` from `AgentCapabilities` in `initialize()`
- Delete the dead `close_session()` handler method from `scripts/ai-sidecar/gemini/sidecar.py`
- Remove unused imports: `CloseSessionResponse`, `SessionCapabilities`, `SessionCloseCapabilities`
- Preserve cleanup behavior via the existing `prompt()` `finally` block (lines 432-440), which already removes `_contextual_tools` for all sessions and prevents unbounded state growth

## Checklist
- [x] I have read https://github.com/jaegertracing/jaeger/blob/main/CONTRIBUTING_GUIDELINES.md
- [x] I have signed all commits
- [ ] I have added unit tests for the new functionality
- [x] I have run lint and test steps successfully: `make lint test`

## AI Usage in this PR
- [ ] None
- [x] Light: AI provided minor assistance (formatting, PR description)

### CODE REVIEW

File: scripts/ai-sidecar/gemini/sidecar.py [hunk 1]
```diff
@@ -154,21 +149,6 @@ async def new_session(
...
-        close again. Capability is advertised in ``initialize`` via
-        ``SessionCapabilities.close``.
-        """
-        self._contextual_tools.pop(session_id, None)
-        logger.info("Closed session %s", session_id)
-        return CloseSessionResponse()
-
     async def load_session(
         self,
         cwd: str,
```
[the remark this answers is not in the document]
Comment (Nabil-Salah) [reply scripts/ai-sidecar/gemini/sidecar.py:160]: we don't enforce agents to support this capibility in jaeger backend.

**Comment by yurishkuro** [comment]
how do we test that this is not leaking memory?

**Comment by qsxDree** [comment]
> how do we test that this is not leaking memory?

by leaking memory. there is a small edge case where is a ```new_session``` starts but the ```prompt``` never actually runs then that session's tools just sit in ```_contextual_tools``` until the WebSocket connection closes.

if you want, there is a way to completely eliminate this edge case, by ditching the dictionary entirely and just use a single variable/slot for the tools. since Jaeger only ever handles one active session per connection anyway, ```new_session``` simply overwrites whatever is currently in that slot and ```prompt``` reads it, and then clears it out when finished.

would this be a reasonable approach?
cc: @yurishkuro

**Review by yurishkuro — COMMENTED** [review]
## Review

This PR removes the `SessionCapabilities.close` advertisement from the sidecar's `initialize()` response and deletes the `close_session()` handler that was never actually being reached (because the Go gateway's `run_agent()` call uses stable-protocol mode, so `session/close` was dispatched as `method_not_found` before even reaching the handler). The fix is conceptually sound: the Go gateway has an explicit capability gate in `closeACPSession()` that skips sending `session/close` entirely when the capability is absent. Two findings below.

---

### 1. Missing regression test

The existing `test_sidecar_workflow.py` does invoke the initialize flow but (a) only asserts `protocolVersion` and `agentInfo.name` — never `agentCapabilities` — and (b) the test's `FakeAgent.initialize()` hard-codes `agent_capabilities=AgentCapabilities()` anyway, bypassing the real sidecar's capability construction entirely. If someone re-adds `SessionCapabilities` to `sidecar.py`'s `initialize()`, the full test suite passes without warning.

A one-line assertion in a test that exercises the real sidecar's `initialize()` path would suffice:
```python
assert response.agent_capabilities == AgentCapabilities()
```

---

### 2. Orphaned sessions with contextual tools leak `_contextual_tools` entries

The deleted `close_session()` was the only cleanup path for sessions that receive contextual tools via `new_session()` but are abandoned before `prompt()` is called (connection drop, timeout, client abort during setup). The `prompt()` `finally` block only runs when `prompt()` actually executes. `_contextual_tools` has no TTL, no max-size guard, and no periodic sweep.

The leak is narrower than it sounds: entries only appear when `_extract_contextual_tools()` returns non-empty, and issue #8738 itself acknowledges "one stale contextual-tools entry per orphan session until process restart" as an accepted tradeoff. That's reasonable. But please add a short comment in `new_session()` noting the orphan-session leak so the next maintainer doesn't re-add `close_session()` without understanding why it was removed.

**Comment by qsxDree** [comment]
Hi @yurishkuro addressed the review
